NEXT-29107 - Add docs for B2B components' entities (#1007)
shopware/docs: products/extensions/b2b-components/employee-management/concept/README.md

# Concept

## Basic idea

> Employee management, as part of the B2B components, is a feature that allows you to manage employees and their permissions as an extension to Shopware's account and customer management, but gathered in a company context. This means that employees are associated with a **company customer** and will act on behalf of that company, when talking about e.g. placing orders in the shop using addresses and payment methods defined by administrators of such company.

Reviewer: ```suggestion
Employee management, as part of the B2B components, is a feature that allows you to manage employees and their permissions as an extension to Shopware's account and customer management, but gathered in a company context. This means that employees are associated with a **company customer** and will act on behalf of that company when, for example, placing orders using addresses defined by administrators of such company.
```

Let's keep the payment methods out of this for now, otherwise it might be misleading regarding the current feature set.

> For that matter, we needed a way to provide order relevant data without building the checkout process anew and decided on using a **company customer**

Reviewer: ```suggestion
For that matter, we needed a way to inject company managed data into core processes, without building them from scratch and without maintaining multiple versions of the same processes. This is why we decided on using a **company customer**.
```

Author: Meyer and me created a new approach there. I guess that fits. Please check again.

## Company customer

> The company customer is a regular customer in the shop, but with a few additional properties. This customer's ID is used to associate employees with a company. Therefore, it is really easy to relate on Shopware's typical order relevant data like addresses and payment methods.

Reviewer: ```suggestion
The company customer is a regular storefront customer, but with a few additional properties. This customer's ID is used to associate employees with a company. Therefore, it is really easy to relate on Shopware's typical order relevant data like addresses and payment methods.

Employees are associated with the same company customer, but are determined by different login credentials.

```

> Internally, every employee will act like it is the company customer itself, using mostly the default handling of Shopware's Core.

Reviewer: ```suggestion
This allows us to keep using most core implementations, while only extending a few B2B related features.
```

> The entities of the B2B components will persist additional like the name of the employee in addition for internal purposes like the distinction of employee actions.

Reviewer: I guess we can drop this line, since we have the entity overview already. Otherwise, please re-read and re-phares it 😄

## Role management

> Employees can be assigned roles that define their permissions and certain settings. These permissions can restrict or allow employees to perform certain actions, like ordering or managing roles or employees themselves. Learn how permissions can be extended [via plugin](products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-plugin.md) or [via app](products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-app.md) in our guides section.

Reviewer 2: Is the `via app` part still valid?

Reviewer 3: It is outdated, the docs entry needs also to be removed

Reviewer: There will be changes to it, but afaik it will still be possible for apps to extend permissions.

Author: But is this scope of this PR? I think that's up to the other guys, isn't it?

Reviewer: Not scope of this PR, just leave this section as it is

> Employees can be assigned roles that define their permissions and certain settings. These permissions can restrict or allow employees to perform certain actions, like ordering or managing roles or employees themselves. Learn how permissions can be extended [via plugin](products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-plugin.md) or [via app](products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-app.md) in our guides section.

Reviewer: ```suggestion
Roles define permissions and certain feature settings for employees. These permissions can restrict or allow certain actions, like placing orders or managing roles or employees themselves. It is supposed to provide an entrypoint for extensions, as B2B requirements heavily depend on your individual projects. 

Learn how permissions can be extended [via plugin](products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-plugin.md) or [via app](products/extensions/b2b-components/employee-management/guides/creating-own-permissions-via-app.md) in our guides section.
```

Author: We re-wrote this section. Do you think its fine now?
